### ISSUE TITLE
Hide OBS from screen capture issue after minimizing

### ISSUE THREAD

**Issue by Bolthier** [obsproject/obs-studio#6263]
### Operating System Info

Windows 10

### OBS Studio Version

27.2.4

### OBS Studio Log URL

https://obsproject.com/logs/Eky26pHWsegfdKFy

### Expected Behavior

When bringing OBS back from minimized or tray back to the front the setting to hide OBS should apply if activated.

### Current Behavior

When using the setting to hide OBS from getting captured, reactivating OBS back to the front results in a black capture screen or OBS getting captured. Depending on if OBS was minimized to tray or just to the background.

### Steps to Reproduce

Minimizing OBS to background
1. Go to settings. General > General > Check: Hide OBS windows from screen capture
2. See no OBS on captured screen
3. Minimize OBS
4. Bring OBS back to front
5. See OBS on captured screen
*Screenshot 1*
![Screenshot 2022-04-01 18-09-27](https://user-images.githubusercontent.com/42382626/161301535-8dcfac7b-6cb1-4d3e-b5f3-b2e0c148bcd6.png)

Minimizing OBS to tray
1. Go to settings. General > General > Check: Hide OBS windows from screen capture
2. See no OBS on captured screen
3. Minimize OBS to tray > Click on tray icon or rightclick hide
4. Bring OBS back to front > Click on tray icon or rightclick show
5. See normal capture screen
6. Repeat steps 3 and 4
7. See black capture screen while in OBS and when creating a screenshot
*Screenshot 2*
![Screenshot 2022-04-01 18-16-49](https://user-images.githubusercontent.com/42382626/161303540-21190be4-ce48-458b-9f48-31bc0479f4cc.png)

**Comment by Fenrirthviti** [comment]
I have confirmed both of these instances, but I'm not sure why it happens, or if there's anything we can do on our end about it, since we're using a Windows-provided function to hide the window from capture.

As a side not to this, during testing I've noticed that about half the time changing the hide OBS setting only applies after restarting the application, whereas the other half it will apply immediately after clicking apply. I'm not sure what causes this inconsistency, but others have reported it as well, and it may (?) be related.
